airbyte-integrations/bases/standard-test/standard_test/standard_test.py

```diff
@@ -0,0 +1,84 @@
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+SOFTWARE.
+"""
+
+from collections import Counter
+
+import inflection as inflection
+import pytest
+from airbyte_protocol import Type, Status, ConnectorSpecification
+from .utils import full_refresh_only_catalog, ConnectorRunner
+
+
+@pytest.mark.usefixtures("inputs")
+class BaseTest:
+
+    @classmethod
+    def config_key(cls):
+        class_name = cls.__name__
+        if class_name.startswith("Test"):
+            class_name = class_name[len("Test"):]
+        return inflection.underscore(class_name)
+
+
+class TestCore(BaseTest):
+    def test_check(self, connector_config, docker_runner: ConnectorRunner):
+        output = docker_runner.call_check(config=connector_config)
+        con_messages = [message for message in output if message.type == Type.CONNECTION_STATUS]
+
+        assert len(con_messages) == 1, "Connection status message should be emitted exactly once"
+        assert con_messages[0].connectionStatus.status == Status.SUCCEEDED
+
+    def test_discover(self, connector_config, catalog, docker_runner: ConnectorRunner):
+        output = docker_runner.call_discover(config=connector_config)
+        catalog_messages = [message for message in output if message.type == Type.CATALOG]
+
+        assert len(catalog_messages) == 1, "Catalog message should be emitted exactly once"
+        # assert catalog_messages[0].catalog == catalog, "Catalog should match the one that was provided"
+
+    def test_spec(self, connector_spec, docker_runner: ConnectorRunner):
+        output = docker_runner.call_spec()
+        spec_messages = [message for message in output if message.type == Type.SPEC]
+
+        assert len(spec_messages) == 1, "Spec message should be emitted exactly once"
+        # assert spec_messages[0].spec == spec, "Spec should be equal to the one in spec.json file"
+
+    def test_read(self, connector_config, configured_catalog, docker_runner: ConnectorRunner):
+        configured_catalog = full_refresh_only_catalog(configured_catalog)
+        output = docker_runner.call_read(connector_config, configured_catalog)
+        records = [message.record for message in output if message.type == Type.RECORD]
+        counter = Counter(record.stream for record in records)
+
+        all_streams = set(stream.stream.name for stream in configured_catalog.streams)
+        streams_with_records = set(counter.keys())
+        streams_without_records = all_streams - streams_with_records
+
+        assert not streams_without_records, f"all streams should return some records, streams without records: {streams_without_records}"
```

Comment:
this should be configurable with a boolean flag probably (all streams vs. some streams)